### ISSUE TITLE
Launch: domain upsell repeats twice

### ISSUE THREAD

**Issue by worldomonation** [Automattic/wp-calypso#49894]
### Steps to reproduce the behavior

1. Create or select an unlaunched site.
2. Click on Sidebar > Site > Posts > Add new post.
3. Enter some text in the post.
4. Click on Masterbar > Launch.
5. Once launch flow takes over, click on `Skip Purchase`

#### What I expected to happen
Site is launched with the free .wordpress.com domain after the step.

#### What actually happened
Launch flow repeats again. 
Site is launched once user makes a selection.

### Context
Back to basics user testing.

#### Browser / OS version
OS: macOS Big Sur 11.1 (20C69)
Browser: Safari Version 14.0.2 (16610.3.7.1.9)
Browser: Firefox Nightly 87.0a1

#### Is this specific to the applied theme? Which one?
N/A

#### Does this happen on simple or atomic sites or both?
Tested with Simple.

#### Is there any console output or error text?
No.

#### Level of impact (Does it block purchases? Does it affect more than just one site?)
Minor/Moderate - does not leave user with a good or polished impression.

#### Screenshot / Video: If applicable, add screenshots to help explain your problem.

https://user-images.githubusercontent.com/6549265/107431717-b317de00-6adb-11eb-9471-9eb0b58b7d80.mov

**Comment by ianstewart** [comment]
@worldomonation I tried to reproduce this issue just now and couldn't. Are you still able to reproduce?

**Comment by worldomonation** [comment]
> @worldomonation I tried to reproduce this issue just now and couldn't. Are you still able to reproduce?

I noticed there are now two launch flows:
- one that looks like Gutenboarding.
- one that is the existing flow, shown in the issue video.

I re-tested both flows now and it would appear that it is no longer reproducible, at least in the two runs I made (1 of each flow).

Given that, let's close this as `Closed - Not Reproducible` for the time being, and if I encounter it again I'll re-open this issue and make note of it.

**Comment by worldomonation** [comment]
2021 02 10 Old Launch Flow
https://user-images.githubusercontent.com/6549265/107560532-4f9cb780-6b92-11eb-8bf4-6e182125ef82.mov

2021 02 10 New Launch Flow
https://user-images.githubusercontent.com/6549265/107560673-7d81fc00-6b92-11eb-862d-86c5f6b03ec4.mov

**Comment by worldomonation** [comment]
I was able to reproduce this on a newly created site but I was not able to reproduce it intentionally when attempting to verify my hypothesis.

The hypothesis was that a site is required to have selected a non-Free plan during Gutenboarding and have purchased a domain (sandbox OK). When the `Launch` process is initiated from Post Editor, the existence of the plan and domain causes issues with the upsell.

Unfortunately this hypothesis could not reproduce the issue when I tried with a new site.

Video:
https://user-images.githubusercontent.com/6549265/108312626-e3aae800-716b-11eb-82bf-74a66c38ff1d.mov
